odl/operator/operator.py

```diff
@@ -2165,7 +2165,7 @@ def derivative(self, x):
         if self.is_linear:
             return self
         else:
-            return self.operator.derivative(x) * self.vector
+            return self.operator.derivative(self.vector * x) * self.vector
```

Comment:
Apparently I wasn't fast enough :-)
Please first check if we have to multiply. This is a hot code path and we need to do our best to reduce wasted effort.

Reply:
I improved this by not checking for linearity, but if the derivative coincides with the operator itself. I checked that it works for ``odl.RealPart``, and all the tests pass.

Reply:
After discussion with @olivierverdier: What about setting the ``linear`` flag for the ``RealPart`` to ``True``?

Reply:
I hope, I got everything. Since the ``linear`` flag was used for other stuff in the RealPart and ImagPart operators, I introduced some additional flag for them.